src/RedemptionPool.sol

```diff
@@ -98,74 +128,88 @@ contract RedemptionPool is Ownable {
 
     /// @notice deposit GRO tokens to the pool before the deadline
     /// @param _amount amount of GRO tokens to deposit
-    function deposit(uint256 _amount) external {
-        // Checks that the DEADLINE has not passed
-        if (block.timestamp > DEADLINE) {
-            revert RedemptionErrors.DeadlineExceeded();
-        }
+    function deposit(uint256 _amount) external onlyBeforeDeadline {
+        if (_amount <= 0) revert RedemptionErrors.GreaterThanZeroOnly();
+
         // Transfers the GRO tokens from the sender to this contract
         GRO.safeTransferFrom(msg.sender, address(this), _amount);
         // Increases the balance of the sender by the amount
-        _userBalance[msg.sender] += _amount;
+        _userGROBalance[msg.sender] += _amount;
         // Increases the total deposited by the amount
         totalGRO += _amount;
         // Emits the Deposit event
-        emit Deposit(msg.sender, _amount);
+        emit Deposit(msg.sender, _amount, totalGRO);
     }
 
-    /// @notice withdraw deposited GRO tokens before the deadline
-    /// @param _amount amount of GRO tokens to withdraw
-    function withdraw(uint256 _amount) external {
```

Comment:
Again, see no reason for removing withdraw. If contract is borked, funds will be lost forever

Reply:
Keep but with modifier onlyBeforeDeadline?

Reply:
Yeah, only before deadline